### DISCUSSION TITLE
Upgrade from 1.1.2 to 1.7.0

### DISCUSSION THREAD

**Discussion by Arthurembo** [smallrye/smallrye-graphql#1513] [Q&A]
Hi guys,
is there a guideline for this upgrade?. I need help for this. I found that, in the version 1.1.2 there was a GraphQlClientBuilderImpl class, but there is no more in the version 1.7.0. I don't exactly now how to get the TypesafeGraphQlClientBuilder instanciate, there is no impl and I always get an error like:
`no io.smallrye.graphql.client.typesafe.api.TypesafeGraphQLClientBuilder in classpath` by calling the method `TypesafeGraphQLClientBuilder.newBuilder()`.

**Comment by t1** [comment]
No, there's no such guide, yet. But this discussion could become one ;-)

To answer this specific question: We have split the API from the implementation, so you need the dependency on `smallrye-graphql-client-implementation-vertx`. Then calling `TypesafeGraphQLClientBuilder.newBuilder` should work.

**Reply by Arthurembo** [reply]
Hi @t1,
thank you very much for the answer. I spent a lot time trying to understand how this work. It may be really helpful to have such a documentation, may be a small one with some bullet points about the technical changes.

**Reply by t1** [reply]
I completely understand your requirement, but we are all not getting paid to do this. So we focus on the absolutely most important things. It's open source, so we welcome all comments and are happy about any pull requests that help us to reach our common goal. Right now, it's more important to get the typesafe client into the MicroProfile spec than precisely documenting the differences between all the versions we have already released. It is currently still marked as experimental, but it seems to have matured enough to take this step. There haven't been any breaking changes in the last releases. When it's part of the standard, we will try to introduce absolutely no breaking changes any more, unless inevitably necessary.
